activerecord/test/cases/adapter_test.rb

```diff
@@ -211,6 +211,28 @@ def test_numeric_value_out_of_ranges_are_translated_to_specific_exception
       end
     end
 
+    def test_exceptions_from_notifications_are_not_translated
+      original_error = RuntimeError.new("This RuntimeError shouldn't get translated")
```

Comment:
This test passes even on master, because a `RuntimeError` is never translated: 
https://github.com/rails/rails/blob/v5.1.2/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L622-L623